src/Equinor.ProCoSys.IPO.Infrastructure/Migrations/20210112131850_ConnectSignedByToPerson.cs

```diff
@@ -0,0 +1,194 @@
+﻿using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+namespace Equinor.ProCoSys.IPO.Infrastructure.Migrations
+{
+    public partial class ConnectSignedByToPerson : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.Sql("UPDATE PARTICIPANTS SET SIGNEDBY = NULL");
+
+            migrationBuilder.AlterColumn<string>(
+                name: "LastName",
+                table: "Persons",
+                type: "nvarchar(128)",
+                maxLength: 128,
+                nullable: false,
+                oldClrType: typeof(string),
+                oldType: "nvarchar(64)",
+                oldMaxLength: 64);
+
+            migrationBuilder.AlterColumn<string>(
+                name: "FirstName",
+                table: "Persons",
+                type: "nvarchar(128)",
+                maxLength: 128,
+                nullable: false,
+                oldClrType: typeof(string),
+                oldType: "nvarchar(64)",
+                oldMaxLength: 64);
+
+            migrationBuilder.AlterColumn<int>(
```

Comment:
I guess this work as expected with existing data since you start with "UPDATE PARTICIPANTS SET SIGNEDBY = NULL"?

Reply:
Yes, so it will "ruin" IPOs that have been created when testing, bu it is okay since we are only in test stage, right?

Reply:
It depends. I don't know each corner of business rules. But: If there will exists Invitations which is signed, and the code then expect to find valid data in SignedBy/SignedAt, this migration will create "invalid" test data. Right? If so, You should either make the data valid by either delete invalid Invitations, or insert a dummy SignedBy/SignedAt where needed. U C what I mean?

Reply:
I C what u mean. 
I could delete all signatures and set all invitations to inital stage - however test data would still be messed up, because milestone dates in main are set and linked to IPOs.
Inserting dummy signedBy (id of a user in db) will also result in invalid data, since the dummy user is not invited to all IPOs (but I can do it just so we have someone there).
Any input?

Reply:
Do the best of it. I just try to prevent messed up data (also in test) leading to a developer uses time to bugfix something that is not a bug.
Regarding inconsistence of data between new database (your IPO in Azure) and main data is a bigger problem, which we also have in preservation.

Our DEV IPO talk do DEV onprem, and after first use of SchemaResetter on DEV we're out of sync.
Our TEST IPO talk do QP onprem, and after first clone from Prod to QP we're out of sync.